impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -399,12 +574,64 @@ private EndpointPair buildExposedEndpoint(Listener listener, int nodeId, int bin
                 .build();
     }
 
+    private InspectContainerCmd buildInspectionCommandFor(KafkaContainer kc) {
+        return kc.getDockerClient().inspectContainerCmd(kc.getContainerId());
+    }
+
+    @SuppressWarnings({ "try" })
+    private static String createNamedVolume() {
+        var volumeCmd = DockerClientFactory.lazyClient().createVolumeCmd();
+        if (!CONTAINER_ENGINE_DOCKER) {
+            volumeCmd.withDriverOpts(Map.of("o", "uid=" + KAFKA_CONTAINER_UID));
+        }
+        var volumeName = volumeCmd.exec().getName();
+        volumesPendingCleanup.add(volumeName);
+        if (CONTAINER_ENGINE_DOCKER) {
+            // On Docker, use a container to chown the volume.
+            // This is a workaround for https://github.com/moby/moby/issues/45714
+            try (var c = new OneShotContainer()) {
+                c.withName("prepareKafkaVolume")
+                        .addGenericBind(new Bind(volumeName, new Volume(KAFKA_CONTAINER_MOUNT_POINT)))
+                        .withCommand("chown", "-R", KAFKA_CONTAINER_UID, KAFKA_CONTAINER_MOUNT_POINT)
+                        .withStartupCheckStrategy(new OneShotStartupCheckStrategy());
+                c.start();
+            }
+        }
```

Comment:
The kafka image runs as uid 1001.  The issue that this is working around is the fact that named volume is owned by root, so 1001 will fail to write to it.  The OneShotContainer runs (as root) a chown to 1001 to resolve this problem. On Podman we don't need this step since it lets you specify a uid when you create the named volume.